### PR TITLE
Simplification of the configuration

### PR DESCRIPTION
**Pull request by petervandenhout**
Hi,

I was working on making the provider use the app settings, so you can use it in an azure website and you don't need the config in your code. **cyberprune** was also working on this and already requested a pull request, that you approved today :) But I think I have some additional changes/simplifications that could be nice to have. This is what I've done... (all tests are passing and the branch is up to date with the latest develop branch).

I've removed the following app settings and config parameters:
- AzureBlobFileSystem.UseStorageEmulator
- rootUrl

The `AzureBlobFileSystem.UseStorageEmulator` could be removed since you can get that information from the connectionstring. If it is `UseDevelopmentStorage=true` then the emulator should be used.
The `rootUrl` parameter is not needed since you can get the blob url from the container `this.cloudBlobContainer.StorageUri.PrimaryUri.AbsoluteUri`.

I also updated the `README.md`.

### CODE REVIEW

**Comment by JimBobSquarePants** [comment]
Ah interesting Thanks! @Jeavon Could you look at this while you're looking at the other stuff I did today?

**Comment by olosvik** [comment]
I think this would be a great addon as it simplifies the checkin to source control and continous deployment to Azure. Hopefully it would be added soon!

File: src/UmbracoFileSystemProviders.Azure.Tests/UmbracoFileSystemProviders.Azure.Tests.csproj [hunk 1]
```diff
@@ -276,6 +276,9 @@
     <Analyzer Include="..\packages\StyleCop.Analyzers.1.0.0-beta014\analyzers\dotnet\cs\Newtonsoft.Json.dll" />
     <Analyzer Include="..\packages\StyleCop.Analyzers.1.0.0-beta014\analyzers\dotnet\cs\StyleCop.Analyzers.dll" />
   </ItemGroup>
+  <ItemGroup>
```
Comment (JimBobSquarePants) [inline src/UmbracoFileSystemProviders.Azure.Tests/UmbracoFileSystemProviders.Azure.Tests.csproj]: What's this?

File: README.md [hunk 2]
```diff
@@ -48,48 +48,34 @@ Update `~/Config/FileSystemProviders.config` replacing the default provider with
...
-      <add key="rootUrl" value="http://[myAccountName].blob.core.windows.net/" />
-      <add key="connectionString" value="DefaultEndpointsProtocol=https;AccountName=[myAccountName];AccountKey=[myAccountKey]"/>
-      <!--
-        Optional configuration value determining the maximum number of days to cache items in the browser.
-        Defaults to 365 days.
-      -->
-      <add key="maxDays" value="365" />
-    </Parameters>
-  </Provider>
+  <Provider alias="media" type="Our.Umbraco.FileSystemProviders.Azure.AzureBlobFileSystem, Our.Umbraco.FileSystemProviders.Azure" />
```
Comment (JimBobSquarePants) [inline README.md]: I'd like to keep the existing config guide as-is and add an additional section in the readme for configuration via app settings.

File: src/UmbracoFileSystemProviders.Azure.Tests/UmbracoFileSystemProviders.Azure.Tests.csproj [hunk 3]
```diff
@@ -276,6 +276,9 @@
...
+    <Service Include="{82A7F48D-3B50-4B1E-B82E-3ADA8210C358}" />
```
Comment (JimBobSquarePants) [inline src/UmbracoFileSystemProviders.Azure.Tests/UmbracoFileSystemProviders.Azure.Tests.csproj]: Whats this?

File: src/UmbracoFileSystemProviders.Azure/AzureBlobFileSystem.cs [hunk 4]
```diff
@@ -22,23 +22,53 @@ public class AzureBlobFileSystem : IFileSystem
         /// Initializes a new instance of the <see cref="AzureBlobFileSystem"/> class.
         /// </summary>
         /// <param name="containerName">The container name.</param>
-        /// <param name="rootUrl">The root url.</param>
         /// <param name="connectionString">The connection string.</param>
-        public AzureBlobFileSystem(string containerName, string rootUrl, string connectionString) :
-            this(containerName, rootUrl, connectionString, "365")
+        /// <param name="maxDays">The maximum number of days to cache blob items for in the browser.</param>
```
Comment (JimBobSquarePants) [inline src/UmbracoFileSystemProviders.Azure/AzureBlobFileSystem.cs]: Why did you delete the overload?

File: src/UmbracoFileSystemProviders.Azure/AzureBlobFileSystem.cs [hunk 5]
```diff
@@ -22,23 +22,53 @@ public class AzureBlobFileSystem : IFileSystem
...
+        public AzureBlobFileSystem(string containerName, string connectionString, string maxDays)
         {
+            bool useDevelopmentStorage = false;
+            if (connectionString.Trim().Equals("UseDevelopmentStorage=true", StringComparison.InvariantCultureIgnoreCase))
+            {
+                useDevelopmentStorage = true;
+            }
+
+            this.FileSystem = AzureFileSystem.GetInstance(containerName, connectionString, maxDays, useDevelopmentStorage, false);
```
Comment (JimBobSquarePants) [inline src/UmbracoFileSystemProviders.Azure/AzureBlobFileSystem.cs]: I'm fairly certain this will break everybody's upgrade path if they have chosen to disable the virtual path provider.

**Comment by JimBobSquarePants** [comment]
Hi @petervandenhout Thanks for this. Unfortunately I cannot merge it in it's current state.

The way you have changed the constructors and apply the configuration values breaks the upgrade path for some users. We need to be able to allow existing installs (Everyone on UAAS) to be able to upgrade without issue so the existing configuration format should yield exactly the same result as prior to the changes.

If you are able to rebase the PR and make only the absolute minimum of changes then I will be happy to accept it. If not, then I can cherry pick from your PR and update it myself.

**Comment by petervandenhout** [comment]
Hi @JimBobSquarePants First sorry for the late reply. Maybe it's easier/better that you just cherry pick from this branch and implement it as you see fit. I don't really have much time to make changes and fix the issues at the moment.
